builtin/aws/ecs/task.go

```diff
@@ -0,0 +1,248 @@
+type TaskLauncher struct {
+	config TaskLauncherConfig
+}
+
+// StartTaskFunc implements component.TaskLauncher
+func (p *TaskLauncher) StartTaskFunc() interface{} {
+	return p.StartTask
+}
+
+// StopTaskFunc implements component.TaskLauncher
+func (p *TaskLauncher) StopTaskFunc() interface{} {
+	return p.StopTask
+}
+
+// TaskLauncherConfig is the configuration structure for the task plugin. At
+// this time all these are simply copied from what the Waypoint Server
+// installation is using, with the only exception being the OdrTaskRoleName.
+type TaskLauncherConfig struct {
+	// Cluster is the ECS we're operating in
+	Cluster string `hcl:"cluster,optional"`
+
+	// Region is the AWS region we're operating in, e.g. us-west-2, us-east-1
+	Region string `hcl:"region,optional"`
+
+	// ExecutionRoleName is the name of the AWS IAM role to apply to the task's
+	// Execution Role. At this time we reuse the same Role as the Server
+	// Execution Role.
+	ExecutionRoleName string `hcl:"execution_role_name,optional"`
+
+	// TaskRoleName is the name of the AWS IAM role to apply to the task. This
+	// role determines the privileges the ODR builder has, and must have the correct
+	// policies in place to work with the provided registries.
+	TaskRoleName string `hcl:"task_role_name,optional"`
+
+	// Subnets are the list of subnets for the cluster. These will match the
+	// subnets used for the Cluster
+	Subnets string `hcl:"subnets,optional"`
+
+	// SecurityGroupId is the security group used for the Waypoint tasks.
+	SecurityGroupId string `hcl:"security_group_id,optional"`
+
+	// LogGroup is the CloudWatch log group name to use.
+	LogGroup string `hcl:"log_group,optional"`
+}
+
+func (p *TaskLauncher) Documentation() (*docs.Documentation, error) {
+	doc, err := docs.New(
+		docs.FromConfig(&TaskLauncherConfig{}),
+		docs.FromFunc(p.StartTaskFunc()),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	doc.Description(`
+Launch an ECS task for on-demand tasks from the Waypoint server.
+
+This will use the standard AWS environment variables and IAM Role information to
+source authentication information for AWS, using the configured task role.
+If no task role name is specified, Waypoint will create one with the required
+permissions.
+`)
+
+	doc.SetField(
+		"odr_task_role_name",
+		"Task role name to be used for the task role in the On-Demand Runner task",
+		docs.Summary(`
+This role must have the correct IAM policies to complete it's task.
+If this IAM role does not already exist, a role will be created with the correct
+permissions"
+`,
+		),
+	)
+
+	return doc, nil
+}
+
+// TaskLauncher implements Configurable
+func (p *TaskLauncher) Config() (interface{}, error) {
+	return &p.config, nil
+}
+
+// StopTask signals to docker to stop the container created previously. This
+// method is currently unimplemented.
+func (p *TaskLauncher) StopTask(
+	ctx context.Context,
+	log hclog.Logger,
+	ti *TaskInfo,
+) error {
+	return nil
```

Comment:
Maybe we should do this for the other ODR task plugins too, but I think we should log a warning that the StopTask function is unimplemented.

Reply:
Good idea; added a `log.Warn` here in 687b6222